src/charm.py

```diff
@@ -80,6 +83,19 @@ def dict_to_action_output(d: Dict) -> Dict:
     return ret
 
 
+def config_check_decorator(method):
+    """This decorator puts the charm in Blocked State in log_level is misconfigured."""
+
+    @wraps(method)
+    def config_check(self, *args, **kwargs):
+        if not self._log_level_valid:
+            self.unit.status = BlockedStatus("Bad configuration value for log_level")
+            return
+        return method(self, *args, **kwargs)
+
+    return config_check
```

Comment:
I don't like this approach. It's counter-intuitive to use a decorator to validate a class property. Correct me if I'm wrong but there is only one place where we need to validate the config and that's the `on.config_changed` hook. I suppose that your problem is that other hooks may overwrite the unit state. This is an indication that we handle status setting badly and that we need to refactor it. Having only one function that sets the unit's state is probably a step in the right direction.